app/scripts/lib/transaction/util.ts

```diff
@@ -240,6 +244,15 @@ function validateSecurity(request: AddTransactionRequest) {
     return;
   }
 
+  if (
+    internalAccount.some(
+      ({ address }) =>
+        address?.toLowerCase() === transactionParams.to?.toLowerCase(),
```

Comment:
Minor, but if there is no `to` address, such as a deploy, and the `address` is undefined, then this will silent skip validation.

Can we assume `address` is always set on an account so we instead throw if it's missing?

Reply:
I updated code above to `address.toLowerCase()`, `?` was actually not needed